roles/etcd_migrate/tasks/migrate.yml

```diff
@@ -20,34 +23,62 @@
 - name: Check the etcd v2 data are correctly migrated
   fail:
     msg: "Failed to migrate a member"
-  when: "'finished transforming keys' not in l_etcdctl_migrate.stdout"
+  when: "'finished transforming keys' not in l_etcdctl_migrate.stdout and 'no v2 keys to migrate' not in l_etcdctl_migrate.stdout"
 
-# TODO(jchaloup): start the etcd on a different port so noone can access it
-# Once the validation is done
-- name: Enable etcd member
-  service:
-    name: "{{ l_etcd_service }}"
-    state: started
+- name: Migration message
+  debug:
+    msg: "Etcd migration finished with: {{ l_etcdctl_migrate.stdout }}"
 
-- name: Re-introduce leases (as a replacement for key TTLs)
-  command: >
-    oadm migrate etcd-ttl \
-    --cert {{ etcd_peer_cert_file }} \
-    --key {{ etcd_peer_key_file }} \
-    --cacert {{ etcd_peer_ca_file }} \
-    --etcd-address 'https://{{ etcd_peer }}:2379' \
-    --ttl-keys-prefix {{ item }} \
-    --lease-duration 1h
-  environment:
-    ETCDCTL_API: 3
-  with_items:
-  - "/kubernetes.io/events"
-  - "/kubernetes.io/masterleases"
+# For non-embedded etcd, just start the service as it was previously disabled due to offline migration.
+# Then, re-attach the leases
+- block:
+  - name: Enable etcd member (for non-embedded etcd)
+    service:
+      name: "{{ l_etcd_service }}"
+      state: started
+  - name: Re-introduce leases (as a replacement for key TTLs) for non-embedded etcd
+    command: >
+      oadm migrate etcd-ttl \
+      --cert {{ etcd_peer_cert_file }} \
+      --key {{ etcd_peer_key_file }} \
+      --cacert {{ etcd_peer_ca_file }} \
+      --etcd-address 'https://{{ etcd_peer }}:{{ etcd_client_port }}' \
+      --ttl-keys-prefix {{ item }} \
+      --lease-duration 1h
+    environment:
+      ETCDCTL_API: 3
+    with_items:
+    - "/kubernetes.io/events"
+    - "/kubernetes.io/masterleases"
+  when:
+  - not r_etcd_common_embedded_etcd | bool
+
+# For embeddded etcd, start the etcd daemon locally (to re-attach leases).
+# Then, stop the etcd daemon (so the embedded etcd can start again later)
+- block:
+  - name: Enable etcd member (for embedded etcd)
```

Comment:
Could we just start the master and then connect to etcd on 4001 to perform the TTL migration?